include/seqan3/alignment/matrix/debug_matrix.hpp

```diff
@@ -0,0 +1,391 @@
+     * \param entries The entry values as a flat std::vector <#entry_type>.
+     * \param rows    The number of rows.
+     * \param cols    The number of columns.
+     */
+    debug_matrix(std::vector<entry_type> entries, size_t const rows, size_t const cols)
+        : _matrix{std::move(entries), rows, cols},
+          _sequence1{std::nullopt},
+          _sequence2{std::nullopt}
+    {}
+
+    /*!\brief Construct the matrix out of the *entries*, the *rows*, and the *cols* and the corresponding sequences.
+     * \param entries The entry values as a flat std::vector <#entry_type>.
+     * \param rows    The number of rows.
+     * \param cols    The number of columns.
+     * \param sequence1 The first sequence of the sequence alignment.
+     * \param sequence2 The second sequence of the sequence alignment.
+     */
+    debug_matrix(std::vector<entry_type> entries, size_t const rows, size_t const cols,
+                 sequence1_t sequence1, sequence2_t sequence2)
+        : _matrix{std::move(entries), rows, cols},
+          _sequence1{std::forward<sequence1_t>(sequence1)},
+          _sequence2{std::forward<sequence2_t>(sequence2)}
+    {
+        assert(cols == _sequence1.size() + 1u);
+        assert(rows == _sequence2.size() + 1u);
+    }
+
+    /*!\brief Construct the matrix out of an existing matrix.
+     * \param matrix An alignment matrix; Must model seqan3::detail::Matrix.
+     */
+    debug_matrix(matrix_t matrix)
+        : _matrix{std::forward<matrix_t>(matrix)},
+          _sequence1{std::nullopt},
+          _sequence2{std::nullopt}
+    {}
+
+    /*!\brief Construct the matrix out of an existing matrix and two sequences.
+     * \param matrix An alignment matrix; Must model seqan3::detail::Matrix.
+     * \param sequence1 The first sequence of the sequence alignment.
+     * \param sequence2 The second sequence of the sequence alignment.
+     */
+    debug_matrix(matrix_t matrix, sequence1_t sequence1, sequence2_t sequence2)
+        : _matrix{std::forward<matrix_t>(matrix)},
+          _sequence1{std::forward<sequence1_t>(sequence1)},
+          _sequence2{std::forward<sequence2_t>(sequence2)}
+    {
+        assert(_matrix.cols() == _sequence1.size() + 1u);
+        assert(_matrix.rows() == _sequence2.size() + 1u);
+    }
+    //!\}
+
+    //!\copydoc seqan3::detail::Matrix::rows
+    size_t rows() const noexcept
+    {
+        return _matrix.rows();
+    }
+
+    //!\copydoc seqan3::detail::Matrix::cols
+    size_t cols() const noexcept
+    {
+        return _matrix.cols();
+    }
+
+    //!\copydoc seqan3::detail::debug_matrix::_sequence1
+    sequence1_t const & sequence1() const noexcept
+    {
+        return _sequence1;
+    }
+
+    //!\copydoc seqan3::detail::debug_matrix::_sequence2
+    sequence2_t const & sequence2() const noexcept
+    {
+        return _sequence2;
+    }
+
+    //!\copydoc seqan3::detail::Matrix::at
+    entry_type at(size_t const row, size_t const col) const noexcept
```

Comment:
We have in the alignment coordinate column_index_type and row_index_type as strong types. Could you use them here to make the interface strong?

Reply:
We decided to do this later.